### PR TITLE
Make common FGS and convert to ophyd async

### PR DESCRIPTION
**Pull request by olliesilvester**
Fixes dodal  #[413](https://github.com/DiamondLightSource/dodal/issues/413)

Link to dodal PR (if required): #[422](https://github.com/DiamondLightSource/dodal/pull/422)
(remember to update `setup.cfg` with the dodal commit tag if you need it for tests to pass!)

After this is merged, we should be able to combine the `flyscan_xray_centre` and `panda_flyscan_xray_centre` plans

### To test:
1. Confirm functionality is unchanged

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -589,8 +590,12 @@ def fake_fgs_composite(
         False
     )
```
Comment (rtuck99) [inline tests/conftest.py:592]: I assume this is to be invoked separately from future tests and this commented out code can be removed?

**Review by rtuck99 — APPROVED** [review]
All ok - apart from comment re commented out code

**Review by DominicOram — CHANGES_REQUESTED** [review]
Just the one comment, thanks!

**Review by DominicOram — APPROVED** [review]
Great, thank you!

**Pull request merged**